android/src/com/mapswithme/util/Utils.java

```diff
@@ -289,4 +289,10 @@ public static void navigateToParent(Activity activity)
     else
       NavUtils.navigateUpFromSameTask(activity);
   }
+
+  public static boolean directoryExists(String path)
+  {
+    final File file = new File(path);
+    return (file.exists() && file.isDirectory() && file.canRead());
```

Comment:
@deathbaba Here we only verify that the given directory is accessible. Writeability is checked during migration on KitKat.
@ygorshenin I think, it'd be better to call method "isDirectoryReadable()".